### DISCUSSION TITLE
Web-vault: Unable register account

### DISCUSSION THREAD

**Discussion by MagomeYae** [dani-garcia/vaultwarden#2992] [Q&A]
### Subject of the issue
Sign up enabled, but register return 404

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.26.0
* Web-vault version: v2022.11.2
* Running within Docker: false (Base: Unknown)
* Environment settings overridden: false
* Uses a reverse proxy: false
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: web vault v2022.11.2
* Reverse proxy and version: nginx 1.22.1 (but direct is also produce this problem)
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": false,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "/var/lib/vaultwarden/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "/var/lib/vaultwarden",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "/***/***/***********/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": true,
  "domain": "*****://********.***:****",
  "domain_origin": "*****://********.***:****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "/var/lib/vaultwarden/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "ssrwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": 65536,
  "org_creation_users": "",
  "password_hints_allowed": true,
  "password_iterations": 100000,Request Method: POST
Status Code: 404 Not Found
Remote Address: [::1]:3011
Referrer Policy: same-origin```

### Troubleshooting data
In android offical bitwarden app can register account, maybe path is wrong?
![image](https://user-images.githubusercontent.com/91249055/207853435-684cae09-6615-4acd-a5af-09d402629382.png)
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "/var/lib/vaultwarden/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "/var/lib/vaultwarden/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "/var/lib/vaultwarden/templates",
  "tmp_folder": "/var/lib/vaultwarden/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": 65536,
  "web_vault_enabled": true,
  "web_vault_folder": "/usr/share/webapps/vaultwarden-web",
  "websocket_address": "127.0.0.1",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Steps to reproduce
Open Web-vault, register account.

### Expected behaviour
Register successful.

### Actual behaviour
Show a message, mean 404
```Request URL: http://localhost:3011/identity/accounts/register
Request Method: POST
Status Code: 404 Not Found
Remote Address: [::1]:3011
Referrer Policy: same-origin
```

### Troubleshooting data
In android offical bitwarden app can register account, maybe path is wrong?
![image](https://user-images.githubusercontent.com/91249055/207853435-684cae09-6615-4acd-a5af-09d402629382.png)

**Comment by BlackDex** [comment]
You are using a unsupported web-vault for the Vaultwarden version running.

Either use the main branch code, or revert the web-vault.

**Reply by metronidazole** [reply]
How does one do that? I have the exact same problem and I'm using vaultwarden/server:latest as per the readme.

**Reply by BlackDex** [reply]
If you are using the latest tagged images that contains a working version of the web-vault which is not 2022.11.x.

So you should not have a problem. Thats probably something else then. If you do have that version then you probably did something different.

**Reply by metronidazole** [reply]
> If you are using the latest tagged images that contains a working version of the web-vault which is not 2022.11.x.
> 
> So you should not have a problem. Thats probably something else then. If you do have that version then you probably did something different.

I figured it out. It was a bug with the flatpak version. Registering worked on the browser.

**Comment by plinss** [comment]
FWIW, I'm seeing the same issue with v2022.11.2 of the web vault (with all patches applied) and the current vaultwarden. Also with v2022.12.0 (which seems to accept all the patches for 2022.11.2). Downgrading to 2022.10.2 allows registration to work.

**Reply by BlackDex** [reply]
With which version of Vaultwarden?

**Reply by plinss** [reply]
1.26.0

**Reply by BlackDex** [reply]
Then the same comment applies to you as the one who started the topic. You are using an unsupported web-vault with the version of Vaultwarden. 1.26.0 only supported 2022.10.2 as latest version. If you want the newer web-vault, use the `testing` tagged images, or build from the main branch.

**Reply by plinss** [reply]
Ok, understood.

Is there some indication that I'm missing which versions of the web vault are supported by any given version of vaultwarden (which version of vaultwarden is required for a given web vault version)? (Ideally it'd be great if there's something that can be grabbed by a CI job that I'm using to build debian packages for each)

**Reply by BlackDex** [reply]
A bit of a cumbersome way would be extracting the version from the `Dockerfile.j2` which contains the version supported by the release:
https://github.com/dani-garcia/vaultwarden/blob/d0b53a6a3de61d3352afb8fb11c97e29cefa454a/docker/Dockerfile.j2#L62

So, with every tagged release it will have the supported version there.
We currently do not have anything else tough. Maybe we could build something, haven't thought about it yet in a automatic way. Only some wiki or readme way.

**Reply by baptisterajaut** [reply]
> Then the same comment applies to you as the one who started the topic. You are using an unsupported web-vault with the version of Vaultwarden. 1.26.0 only supported 2022.10.2 as latest version. If you want the newer web-vault, use the `testing` tagged images, or build from the main branch.

So the lastest main release of the web vault is not compatible with the lastest main release of vaultwarden, requiring vaultwarden to be on the tetsing branch to run the main web interface. What? Thanks for the detailled answer, i fixed the issue but the admin diagnosis complain about how the web release is not up to date while vaultwarden is......

**Reply by BlackDex** [reply]
The web-vault builds are async to the server builds.
Also, sometimes, as you have noticed your selfs, there are some server side fixes needed to support the new web-vault.
These fixes are done on main and not instantly transferred to `latest`.

The `testing` tagged images are always in sync with the `main` branch.
The `latest` tagged images are always in sync with the latest tagged version.

Because we use the docker images build by the `bw_web_build` GHA in all the images we build for Vaultwarden, the web-vault will always be one step ahead (most of the time).

Ill see if we somehow can put a supported web-vault version into the building of the server, to only show a newer version if that is larger. For the docker-builds we do not show updates at all, since they are linked.

**Reply by plinss** [reply]
In my case I'm building individual Debian packages for Vaultwarden and the web-vault (with the appropriate patches). This way if there's an updated web-vault I can install it independently. Ideally, I'd like to be able to set the dependencies between these two packages so that compatible updates of the web-vault can just be installed by an `apt upgrade` and I never wind up with an incompatible combination.

My packages are auto-built by a CI/CD system whenever there are new releases of either. I don't mind having to write something to extract the data, but the ideal situation for me would be something in the bw_web_builds repo indicating the minimum version of Vaultwarden that's required for a given web-vault version. (I also completely don't expect you to do anything special for me, but hoping others can benefit from this information as well.) (I'm also happy to make my packages available to the community if people want them.)

**Reply by BlackDex** [reply]
It's hard to indicate a vaultwarden version here in this repo. The main reason is that we do not know the new version yet.
The other way around is a bit more easy. I will look into that when i have a bit more time.

**Reply by stefan0xC** [reply]
It's hard to tell what the minimum required version of Vaultwarden would be if there has been a breaking change in the client and there is no new Vaultwarden release. I mean, the next release (i.e. the release based off the current main branch / testing image) should be compatible with `web-v2022.11.0` and later (until the next breaking change) and it should possibly also compatible with earlier releases of the web vault (which at least I have not tested).

The current release (v1.26.0) would be the recommended version for web vault releases starting with `web-v2022.9.2` up to `web-v2022.10.2` and may also be compatible with older versions. But it's incompatible with `web-v2022.11.0` and newer versions of the web vault, which are only compatible with the main branch / `vaultwarden/server:testing`.

To discern if a new version of Vaultwarden will be compatible with a newer web vault version (and/or other Bitwarden clients) you have to follow the issue tracker/ pull requests of this repository and/or the commit messages on the main branch. You can also follow the development of the Bitwarden clients and mobile repositories respectively to keep up to date with the upcoming changes but since not all changes will be included in their next release it's probably easier to check only their releases and/or wait for people using the testing image and e.g. the beta version of Bitwarden mobile apps to report incompatibilites/crashes.

**Reply by stefan0xC** [reply]
If we know that a new web vault version is incompatible with the latest release of vaultwarden we could indicate that in the release notes of the [bw_web_builds](https://github.com/dani-garcia/bw_web_builds) repository. :thinking:
